dsgrid/registry/registry_auto_updater.py

```diff
@@ -0,0 +1,306 @@
+            context.connection, config.model.dimension_id, original_version
+        )
+        for mapping in self._update_dimension_mappings_with_dimensions(
+            affected, config, original_version
+        ):
+            key = ConfigKey(mapping.model.mapping_id, mapping.model.version)
+            new_mapping = self._dimension_mapping_mgr.update_with_context(mapping, context)
+            assert key not in new_mappings
+            new_mappings[key] = new_mapping
+            logger.info(
+                "Updated dimension mapping %s to %s as a result of dimension update",
+                new_mapping.model.mapping_id,
+                new_mapping.model.version,
+            )
+        for dataset in self._update_datasets_with_dimensions(affected, config, original_version):
+            key = ConfigKey(dataset.model.dataset_id, dataset.model.version)
+            new_dataset = self._dataset_mgr.update_with_context(dataset, context)
+            assert key not in new_datasets
+            new_datasets[key] = new_dataset
+            logger.info(
+                "Updated dataset %s to %s as a result of dimension update",
+                new_dataset.model.dataset_id,
+                new_dataset.model.version,
+            )
+
+        self._update_projects(
+            context,
+            dimensions={ConfigKey(config.model.dimension_id, original_version): config},
+            dimension_mappings=new_mappings,
+            datasets=new_datasets,
+        )
+
+    def _update_dimension_mapping_users(
+        self,
+        config: MappingTableConfig,
+        original_version: str,
+        context: RegistrationContext,
+    ) -> None:
+        self._update_projects(
+            context,
+            dimension_mappings={ConfigKey(config.model.mapping_id, original_version): config},
+        )
+
+    def _update_dataset_users(
+        self,
+        config: DatasetConfig,
+        original_version: str,
+        context: RegistrationContext,
+    ) -> None:
+        self._update_projects(
+            context,
+            datasets={ConfigKey(config.model.dataset_id, original_version): config},
+        )
+
+    def _update_dimension_mappings_with_dimensions(
+        self,
+        affected: dict[RegistryType, list[Any]],
+        dim: DimensionConfig,
+        original_version: str,
+    ) -> list[MappingTableConfig]:
+        mapping_updates: list[MappingTableConfig] = []
+        for model in affected[RegistryType.DIMENSION_MAPPING]:
+            assert isinstance(model, MappingTableModel)
+            updated = False
+            config = self._dimension_mapping_mgr.get_by_id(model.mapping_id, model.version)
+            if (
+                config.model.from_dimension.dimension_id == dim.model.dimension_id
+                and config.model.from_dimension.version == original_version
+            ):
+                config.model.from_dimension.version = dim.model.version
```

Comment:
No checks that the dimension mapping is actually valid for the new version? Does that happen later, when affected datasets are re-submitted to project(s)?